### PR TITLE
Add StringIsNotURL validation

### PR DESCRIPTION
**Pull request by alexhung**
This is to support https://github.com/jfrog/terraform-provider-artifactory/pull/453

### CODE REVIEW

**Comment by alexhung** [comment]
@chb0github ☝🏼

File: validator/validator.go [hunk 1]
```diff
@@ -590,6 +591,44 @@ func StringInSlice(ignoreCase bool, valid ...string) schema.SchemaValidateDiagFu
...
+	if !ok {
+		diags = append(diags, diag.Diagnostic{
+			Severity:      diag.Error,
+			Summary:       "Invalid string",
+			AttributePath: p,
+			Detail:        fmt.Sprintf("expected type of %q to be string", attr.Name),
+		})
+	}
+
+	if v == "" {
```
Comment (chb0github) [inline validator/validator.go:610]: Could you directly use `validation.StringNotEmpty` ?

File: validator/validator.go [hunk 2]
```diff
@@ -590,6 +591,44 @@ func StringInSlice(ignoreCase bool, valid ...string) schema.SchemaValidateDiagFu
...
+			Severity:      diag.Error,
+			Summary:       "Invalid string",
+			AttributePath: p,
+			Detail:        fmt.Sprintf("expected %q url to not be empty, got %v", attr.Name, v),
+		})
+	}
+
+	_, err := url.Parse(v)
+	if err == nil {
+		diags = append(diags, diag.Diagnostic{
```
Comment (chb0github) [inline validator/validator.go:621]: Could you use `validation.IsHttpOrHttps`

**Review by chb0github — APPROVED** [review]
I am going to approve, but my suggestions is to use the existing validators if you can

**Pull request merged**